lib/ron.nix

```
@@ -0,0 +1,724 @@
+{ lib, ... }:
+{
+  /**
+    Converts a RON (Rusty Object Notation) string to Nix values.
+
+    This function parses RON format strings and converts them to equivalent Nix data structures.
+    It supports various RON types including primitives, collections, optionals, enums, structs, and tuples.
+
+    Special RON types are represented as attribute sets with a `_type` field:
+    - Optional values: `{ _type = "ron-optional"; value = ...; }`
+    - Characters: `{ _type = "ron-char"; value = "c"; }`
+    - Enums: `{ _type = "ron-enum"; variant = "..."; value = [...]; }`
+    - Maps: `{ _type = "ron-map"; value = [{ key = ...; value = ...; }, ...]; }`
```

Similarly, but less footgun-y

```suggestion
    - Maps: `{ _type = "ron-map"; attrs = [{ key = ...; value = ...; }, ...]; }`
```